Source/player.cpp

```diff
@@ -3889,131 +3889,100 @@ void __cdecl ValidatePlayer()
 
 void __cdecl ProcessPlayers()
 {
-	int v0; // eax
-	int v1; // eax
-	unsigned char *v2; // ecx
-	char v3; // al
-	int v4; // ebp
-	int *v5; // esi
-	int v6; // eax
-	//int v7; // eax
-	int v8; // eax
-	int v9; // eax
-	int v10; // eax
-	int v11; // edi
-	int v12; // eax
-	char *v13; // eax
-	char *v14; // eax
-
-	v0 = myplr;
-	if ( (unsigned int)myplr >= 4 )
-	{
+	if ( (DWORD)myplr >= MAX_PLRS ) {
 		TermMsg("ProcessPlayers: illegal player %d", myplr);
-		v0 = myplr;
 	}
-	v1 = v0;
-	v2 = &plr[v1].pLvlLoad;
-	v3 = plr[v1].pLvlLoad;
-	if ( v3 )
-		*v2 = v3 - 1;
-	v4 = 0;
-	if ( sfxdelay > 0 && !--sfxdelay )
-		PlaySFX(sfxdnum);
+
+	if ( plr[myplr].pLvlLoad ) {
+		plr[myplr].pLvlLoad = plr[myplr].pLvlLoad - 1;
+	}
+
+	if ( sfxdelay > 0 ) {
+		sfxdelay--;
+		if ( sfxdelay == 0 ) {
+			PlaySFX(sfxdnum);
+		}
+	}
+
 	ValidatePlayer();
-	v5 = &plr[0]._pHitPoints;
-	do
-	{
-		v6 = (int)(v5 - 89);
-		if ( *((_BYTE *)v5 - 379) && currlevel == *(_DWORD *)v6 && (v4 == myplr || !*(_BYTE *)(v6 + 267)) )
-		{
-			CheckCheatStats(v4);
-			//_LOBYTE(v7) = PlrDeathModeOK(v4);
-			if ( !PlrDeathModeOK(v4) && (signed int)(*v5 & 0xFFFFFFC0) <= 0 )
-				SyncPlrKill(v4, -1);
-			if ( v4 == myplr )
-			{
-				if ( v5[5294] & 0x40 && currlevel )
-				{
-					*v5 -= 4;
-					v8 = *v5;
-					*(v5 - 2) -= 4;
-					if ( (signed int)(v8 & 0xFFFFFFC0) <= 0 )
-						SyncPlrKill(v4, 0);
+
+	for ( int pnum = 0; pnum < MAX_PLRS; pnum++ ) {
+		if ( plr[pnum].plractive && currlevel == plr[pnum].plrlevel && (pnum == myplr || !plr[pnum]._pLvlChanging) ) {
+			CheckCheatStats(pnum);
+
+			if ( !PlrDeathModeOK(pnum) && (plr[pnum]._pHitPoints >> 6) <= 0 ) {
+				SyncPlrKill(pnum, -1);
+			}
+
+			if ( pnum == myplr ) {
+				if ( plr[pnum]._pIFlags & ISPL_DRAINLIFE ) {
+					plr[pnum]._pHitPoints -= 4;
+					plr[pnum]._pHPBase -= 4;
+					if ( (plr[pnum]._pHitPoints >> 6) <= 0 ) {
+						SyncPlrKill(pnum, 0);
+					}
 					drawhpflag = 1;
 				}
-				if ( *((_BYTE *)v5 + 21179) & 8 )
-				{
-					v9 = v5[3];
-					if ( v9 > 0 )
-					{
-						v10 = v9 - v5[5];
-						v5[5] = 0;
-						drawmanaflag = 1;
-						v5[3] = v10;
-					}
+				if ( plr[pnum]._pIFlags & ISPL_NOMANA && plr[pnum]._pManaBase > 0 ) {
+					plr[pnum]._pManaBase -= plr[pnum]._pMana;
+					plr[pnum]._pMana = 0;
+					drawmanaflag = 1;

```

Comment:
Is `drawmanaflag` a boolean, if so use TRUE instead of 1.

Reply:
drawmanaflag is being used as a bool in all cases, the same goes for drawhpflag, both are defined as int, but I agree, they should probably be changed to BOOL. There are other similar types used in DrawAndBlit()

Reply:
I have changed bot to BOOLs now.